### PR TITLE
[OSF-8228] fix navbar responsiveness for long usernames with wide characters

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Currently the navbar is responsive and collaspes well for long usernames, but when those names consist of wide characters (like uppercases Ms) the responsiveness still breaks down into two levels. 

## Before
before_navbar_fix

## After
after_navbar_fix

Also:

There's a small gray area creeping out from under the blue highlighted area on project navbar, this PR fixes that.

## Before 
screen shot 2017-12-18 at 11 10 53 am

You can see the small strip of gray under the project name "Private ?"

## After
screen shot 2017-12-18 at 11 08 57 am
 Now it's gone.

## Changes

- Changes the HTML to better accommodate responsiveness css changes will be included in osf-style.
- Should match the above after gif.

## QA Notes

- For usernames use wide characters and long usernames with and out spaces.
- Test on a variety of browsers.

## Deployment Notes

This needs corresponding PR in osf-style to be merged in order to work.
https://github.com/CenterForOpenScience/osf-style/pull/109

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/OSF-8228

### CODE REVIEW

File: website/templates/nav.mako [hunk 1]
```diff
@@ -48,19 +48,22 @@
...
-                    <img src="${user_profile_image}" alt="User profile image">
-                </span> ${display_name}
-                <span class="caret"></span>
-            </a>
+                <a class="dropdown-toggle btn-link" data-toggle="dropdown" role="button" aria-expanded="false" aria-label="Toggle auth dropdown">
+                    <div class="osf-profile-image">
+                        <img src="${user_profile_image}" alt="User profile image">
+                    </div>
+                    <div class="nav-profile-name">
+                            ${display_name}
```
Comment (jamescdavis) [inline website/templates/nav.mako]: Truncating the display name with CSS obviates the need to truncate the actual value here. If the value is truncated, but still too long, you could end up with something weird like `This is a Very...Long Name That...`

Comment (jamescdavis) [reply website/templates/nav.mako]: I'm not sure if there is a way to easily get the non-truncated Display Name.

Comment (Johnetordoff) [reply website/templates/nav.mako]: I've stopped the truncation, this whole time I didn't know we did this server-side.

**Comment by Johnetordoff** [comment]
@jamescdavis not sure this is actually on me, here's a similar failure on the same branch in a different PR. https://travis-ci.org/CenterForOpenScience/osf.io/jobs/317833028

**Comment by jamescdavis** [comment]
Indeed! Even master is failing with this error: https://travis-ci.org/CenterForOpenScience/osf.io/jobs/318427642

**Pull request merged**